matrix/matrix_operation.py

```diff
@@ -33,9 +44,11 @@ def subtract(matrix_a: list[list], matrix_b: list[list]) -> list[list]:
         and _verify_matrix_sizes(matrix_a, matrix_b)
     ):
         return [[i - j for i, j in zip(*m)] for m in zip(matrix_a, matrix_b)]
+    else:
+        return [[]]
 
 
-def scalar_multiply(matrix: list[list], n: int) -> list[list]:
+def scalar_multiply(matrix: Iterable[list], n: float) -> list[list]:
```

Comment:
I don't think a matrix should _just_ be an `Iterable` type. `list` seems to be an appropriate type for it. And if we do plan to just keep it `Iterable` then it should be the same where we require and return a matrix.